yetanotherco/aligned_layer
# feat: batcher

# Description

Created a new cli tool, that acts as an intermediary between the task sender & the smart contract. It

1. takes proofs from a test client using a web socket connection.
2. Verifies the proof
3. Hashes the data & returns the hash to the client
4. Posts batches to an S3 bucket

# To test

1. Create a .env file inside the batcher directory (e.g at `batcher/.env.example`). As an alternative you can manually set the aws environment variables as described [here](https://docs.aws.amazon.com/sdkref/latest/guide/environment-variables.html)
2. Run `make batcher_start`
3. Run `make batcher_send_dummy_task` twice
4. See logs to check that data is posted


Closes https://github.com/yetanotherco/aligned_layer/issues/160
